src/components/Paginator.res

```diff
@@ -1,60 +1,138 @@
+external formEventToInt: ReactEvent.Form.t => int = "%identity"
+
+let dropdownStyle = "absolute bottom-0 z-50 w-fit mb-11 bg-white dark:bg-jp-gray-950 divide-y divide-gray-100 rounded-md shadow-md ring-1 ring-black ring-opacity-5 focus:outline-none"
+
+module TransitionWrapper = {
+  open HeadlessUI
+  @react.component
+  let make = (~children) => {
+    <Transition
+      \"as"="span"
+      enter="transition ease-out duration-100"
+      enterFrom="transform opacity-0 scale-95"
+      enterTo="transform opacity-100 scale-100"
+      leave="transition ease-in duration-75"
+      leaveFrom="transform opacity-100 scale-100"
+      leaveTo="transform opacity-0 scale-95">
+      {children}
+    </Transition>
+  }
+}
+
+module NewPagination = {
+  @react.component
+  let make = (~resultsPerPage, ~totalResults, ~currentPage, ~paginate) => {
+    let pageNumbers = []
+    let (arrow, setArrow) = React.useState(_ => false)
+    let total = Math.ceil(Int.toFloat(totalResults) /. Int.toFloat(resultsPerPage))->Float.toInt
+
+    for x in 1 to total {
+      Array.push(pageNumbers, x)->ignore
+    }
+
+    let navigationBtnClass = "flex gap-2 items-center w-full p-2 text-sm  hover:bg-gray-100 cursor-pointer"
+
+    let prevBtn =
+      <div
+        onClick={_evt => paginate(Math.Int.max(1, currentPage - 1))} className=navigationBtnClass>
+        <Icon name="chevron-left" size=12 />
+        <div> {"Prev"->React.string} </div>
+      </div>
+
+    let nextBtn =
+      <div onClick={_evt => paginate(currentPage + 1)} className=navigationBtnClass>
+        <div> {"Next"->React.string} </div>
+        <Icon name="chevron-right" size=12 />
+      </div>
+
+    open HeadlessUI
+    <div className="bg-white border rounded-lg font-medium flex">
+      <RenderIf condition={currentPage > 1}> {prevBtn} </RenderIf>
+      <Menu \"as"="div" className="relative inline-block text-left">
+        {_menuProps =>
+          <div>
+            <Menu.Button
+              className="inline-flex whitespace-pre leading-5 justify-center text-sm  px-2 py-2 font-medium hover:bg-gray-100">
+              {_buttonProps => {
+                <>
+                  {`${currentPage->Int.toString} of ${pageNumbers
+                    ->Array.length
+                    ->Int.toString}`->React.string}
+                  <Icon
+                    className={arrow
+                      ? `rotate-0 transition duration-[250ms] ml-1 mt-1 opacity-60`
+                      : `rotate-180 transition duration-[250ms] ml-1 mt-1 opacity-60`}
+                    name="arrow-without-tail"
+                    size=15
+                  />
+                </>
+              }}
+            </Menu.Button>
+            <TransitionWrapper>
+              <Menu.Items className=dropdownStyle>
+                {props => {
+                  if props["open"] {
+                    setArrow(_ => true)
```

Comment:
please check whether your getting any warning in console.Because setState is happening at every render